tests/integration.rs

```diff
@@ -798,6 +825,88 @@ mod tests {
+                        Ok((mut stream, addr)) => {
+                            // Make sure the connection is from the expected 'guest_addr'
+                            if addr.ip() != std::net::IpAddr::from_str(expected_guest_addr).unwrap()
+                            {
+                                s = format!("Expecting the guest ip '{}' while being connected with ip '{}'",
+                                            expected_guest_addr, addr.ip());
+                                return Err(Error::WrongGuestAddr);
+                            }
+
+                            // Make sure the right message is to notify the guest VM is booted
+                            let mut data = String::new();
+                            stream
+                                .set_read_timeout(Some(std::time::Duration::new(
+                                    DEFAULT_HTTP_SERVER_TIMEOUT as u64,
+                                    0,
+                                )))
+                                .map_err(Error::SetReadTimeout)?;
+                            stream
+                                .read_to_string(&mut data)
+                                .map_err(Error::ReadToString)?;
+                            if data != DEFAULT_HTTP_SERVER_MESSAGE {
+                                s = format!("Expecting the guest message '{}' while receiving the message '{}'",
+                                            DEFAULT_HTTP_SERVER_MESSAGE, data);
+                                return Err(Error::WrongGuestMsg);
+                            };
+
+                            Ok(())
+                        }
+                        Err(e) => {
+                            s = "TcpListener::accept() failed".to_string();
+                            Err(Error::Connection(e))
+                        }
+                    }
+                })() {
+                    Ok(_) => break,
+                    Err(e) => {
+                        counter += 1;
+                        if counter >= DEFAULT_HTTP_SERVER_RETRIES || !retry {
+                            eprintln!(
+                                "\n\n==== Start 'wait_vm_boot' (FAILED) ====\n\n\
+                                 listen_addr=\"{}\"\n\
+                                 expected_guest_addr=\"{}\"\n\
+                                 message =\"{}\"\n\
+                                 \n==== End 'wait_vm_boot' outout ====\n\n",
+                                listen_addr, expected_guest_addr, s,
+                            );
+
+                            return Err(e);
+                        }
+                    }
+                };
+                thread::sleep(std::time::Duration::new(
+                    (DEFAULT_HTTP_SERVER_WAIT).into(),
+                    0,
+                ));
+            }
+
+            let duration = start.elapsed();
+            println!("[DBG] wait_vm_boot takes {:?}\n", duration);
```

Comment:
Cool, let's include something like this as an `eprintln!` so if the test fails we'll see it.